### DISCUSSION TITLE
How to ignore words in js and golang import statements ?

### DISCUSSION THREAD

**Discussion by ygqygq2** [streetsidesoftware/vscode-spell-checker#3088] [Q&A] [answered]
My settings like this, but not work in js file.
```
  "cSpell.ignoreRegExpList": [
    "^import.*$",
  ],
```

**Comment by Jason3S** [comment]
@ygqygq2,

Please try `"/^import\\b.*$/gm"` instead.

**Reply by ygqygq2** [reply]
Thanks. This matches import as well.

  "cSpell.ignoreRegExpList": [
    "/^import\\b.*$/gm",
    "/^import\\s+(?:\\w+\\s+)?\"[^\"]+\"|^import \\(\\s*(?:(?:\\w+\\s+)?\"[^\"]+\"\\s*)+\\)/gm"
  ],

**Reply by Jason3S** [reply, marked as answer]
@ygqygq2,

The important part was the `/…/gm`.

I suggest using [regex101: build, test, and debug regex](https://regex101.com/) to test out your expressions.

<img width="1193" alt="image" src="https://github.com/streetsidesoftware/vscode-spell-checker/assets/3740137/fb59cac9-9fce-47fe-8788-a6a5b8747ea8">

I also suggest defining patterns, see [typescript/cspell-ext.json](https://github.com/streetsidesoftware/cspell-dicts/blob/9e844905244ac4eba6c3bd4c2ba569a073c5d419/dictionaries/typescript/cspell-ext.json#L19) and using `languageSettings` to select the pattern for each programming language.

**Reply by ygqygq2** [reply]
Oh, thanks, it ont only a string, is a regex.
